### PR TITLE
feat: build ubi image

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Build UBI based image

This change require to change corresponding uid in Sumo Logic Kubernetes Collection as well as I change uid from 999 to 1000 to avoid removing users and groups from base images

Preflight check:
```
preflight check container docker.io/sumodrosiek/test:setup-0 --platform=arm64
time="2024-05-27T10:17:50+02:00" level=info msg="certification library version" version="0.0.0 <commit: 354d10d9af8b5a64c7c2230a6656186957c3ed9f>"
time="2024-05-27T10:17:51+02:00" level=info msg="running checks for docker.io/sumodrosiek/test:setup-0 for platform arm64"
time="2024-05-27T10:17:51+02:00" level=info msg="target image" image="docker.io/sumodrosiek/test:setup-0"
time="2024-05-27T10:18:46+02:00" level=info msg="check completed" check=HasLicense result=PASSED
time="2024-05-27T10:18:46+02:00" level=info msg="check completed" check=HasUniqueTag result=PASSED
time="2024-05-27T10:18:46+02:00" level=info msg="check completed" check=LayerCountAcceptable result=PASSED
time="2024-05-27T10:18:46+02:00" level=info msg="check completed" check=HasNoProhibitedPackages result=PASSED
time="2024-05-27T10:18:46+02:00" level=info msg="check completed" check=HasRequiredLabel result=PASSED
time="2024-05-27T10:18:46+02:00" level=info msg="USER setup specified that is non-root"
time="2024-05-27T10:18:46+02:00" level=info msg="check completed" check=RunAsNonRoot result=PASSED
time="2024-05-27T10:19:24+02:00" level=info msg="check completed" check=HasModifiedFiles result=PASSED
time="2024-05-27T10:19:26+02:00" level=info msg="check completed" check=BasedOnUbi result=PASSED
time="2024-05-27T10:19:26+02:00" level=info msg="This image's tag setup-0 will be paired with digest sha256:43a42491e155ae22c352f77fd4b9d5d2470acb886b61eb2ba280ce2a659dc4b4 once this image has been published in accordance with Red Hat Certification policy. You may then add or remove any supplemental tags through your Red Hat Connect portal as you see fit."
{
    "image": "docker.io/sumodrosiek/test:setup-0",
    "passed": true,
    "test_library": {
        "name": "github.com/redhat-openshift-ecosystem/openshift-preflight",
        "version": "0.0.0",
        "commit": "354d10d9af8b5a64c7c2230a6656186957c3ed9f"
    },
    "results": {
        "passed": [
            {
                "name": "HasLicense",
                "elapsed_time": 0,
                "description": "Checking if terms and conditions applicable to the software including open source licensing information are present. The license must be at /licenses"
            },
            {
                "name": "HasUniqueTag",
                "elapsed_time": 0,
                "description": "Checking if container has a tag other than 'latest', so that the image can be uniquely identified."
            },
            {
                "name": "LayerCountAcceptable",
                "elapsed_time": 0,
                "description": "Checking if container has less than 40 layers.  Too many layers within the container images can degrade container performance."
            },
            {
                "name": "HasNoProhibitedPackages",
                "elapsed_time": 120,
                "description": "Checks to ensure that the image in use does not include prohibited packages, such as Red Hat Enterprise Linux (RHEL) kernel packages."
            },
            {
                "name": "HasRequiredLabel",
                "elapsed_time": 0,
                "description": "Checking if the required labels (name, vendor, version, release, summary, description) are present in the container metadata."
            },
            {
                "name": "RunAsNonRoot",
                "elapsed_time": 0,
                "description": "Checking if container runs as the root user because a container that does not specify a non-root user will fail the automatic certification, and will be subject to a manual review before the container can be approved for publication"
            },
            {
                "name": "HasModifiedFiles",
                "elapsed_time": 37675,
                "description": "Checks that no files installed via RPM in the base Red Hat layer have been modified"
            },
            {
                "name": "BasedOnUbi",
                "elapsed_time": 2242,
                "description": "Checking if the container's base image is based upon the Red Hat Universal Base Image (UBI)"
            }
        ],
        "failed": [],
        "errors": []
    }
}
time="2024-05-27T10:19:28+02:00" level=info msg="Preflight result: PASSED"
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -33,7 +33,18 @@ RUN cd /monitors/ \
...
+ ARG BUILD_TAG=latest
+ ENV TAG=$BUILD_TAG
+ 
+ LABEL name="Sumo Logic Kubernetes Setup" \
+     vendor="Sumo Logic" \
+     version="${BUILD_TAG}" \
+     release="1" \
+     summary="$SUMMARY" \
+     description="$DESCRIPTION" \
+     maintainer="opensource-collection-team@sumologic.com"
```
Comment (swiatekm-sumo) [inline Dockerfile:45]: I was under the impression these were openshift-specific labels. Why are we adding them to the non-ubi Dockerfile?

Comment (sumo-drosiek) [reply Dockerfile:45]: I can remove them. I think most of them are informative

File: Dockerfile [hunk 2]
```diff
@@ -35,7 +33,18 @@ RUN cd /monitors/ \
...
+ 
+ ADD LICENSE \
+     /licenses/LICENSE
```
Comment (sumo-drosiek) [inline Dockerfile:48]:
```suggestion
```

File: Dockerfile [hunk 3]
```diff
@@ -35,7 +33,8 @@ RUN cd /monitors/ \
  && terraform init -input=false || terraform init -input=false -upgrade \
  && rm -rf terraform-sumologic-sumo-logic-monitor
 
-ARG BUILD_TAG=latest
-ENV TAG=$BUILD_TAG
+ ARG BUILD_TAG=latest
+ ENV TAG=$BUILD_TAG
+
```
Comment (sumo-drosiek) [inline Dockerfile:38]:
```suggestion
```

File: Dockerfile (hunk 1 above)
Comment (sumo-drosiek) [reply Dockerfile]: removed

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**